Dominik48N/party-system
# Server Blacklist

### Changes Made

- add test for list support in Document
- add ServerSwitchConfigTest
- implement ServerSwitchConfig in ProxyPluginConfig
- create base class for SwitchServer
- add check for black & white list server
- supply serverSwitchConfig for listeners

### Reason for Changes

Include a config entry that allows servers to be blacklisted so that if the party leader joins this server, the party is not pulled along.

It would make sense to support explicit server names, as well as the beginning of a server name. As an example, all servers whose names begin with "Lobby" are directly blacklisted.

### Issues Resolved

Now possible to configurate in config Black and Whitelist for server swapping with enable option

### Checklist

- [X] *(Optional)* I have updated the tests accordingly
- [X] My code follows the established code style guidelines
- [X] I have added/updated necessary comments to my code
